### DISCUSSION TITLE
docker: invalid reference format: repository name (library/PGP) must be lowercase

### DISCUSSION THREAD

**Discussion by Apamobil** [renovatebot/renovate#31635] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Self-hosted Renovate 37.388.1-full with Jenkins, Bitbucket

### Please tell us more about your question or problem

I'm running Renovate with docker setup (please find the Jenkins pipeline renovate step below). 

When trying to pass the private key for commit signing into the docker with switch -e RENOVATE_GIT_PRIVATE_KEY=$RENOVATE_GIT_PRIVATE_KEY it fails with error notification "docker: invalid reference format: repository name (library/PGP) must be lowercase". The key has been prepared as described in the Renovate's documentation (Replaced the newlines with \n before adding the resulting single-line value to bot's config etc.)

When running renovate with debug logs enabled it also throws the whole private key into the logs without masking it. In the logs the following is visible (please note I replaced a part of the key with <the full key is printed out here> block since I don't want to expose the key here) :

"
...LOG_LEVEL=debug -e RENOVATE_GIT_PRIVATE_KEY=-----BEGIN PGP PRIVATE KEY 'BLOCK-----\n\nlTcAG<the full key is printed out here>=6XPt\n-----END' PGP PRIVATE KEY BLOCK----- -v ./config.js:/usr/src/app/config.js ZZZ.YYY.com/thisisafolder/renovate/renovate:37.388.1-full --dry-run=false
docker: invalid reference format: repository name (library/PGP) must be lowercase.
See 'docker run --help'.
"

Can you propose what could be causing this? By googling I found out that this could refer to docker's problems in coping with spaces in a parameter/variable. There are spaces in e.g. the begin and end tags of the PGP key block. However, if I fiddle too much with the PGP key it will become invalid... and on the other hand the very same key worked well in the non-dockerized setup of Renovate.

The Pipeline's Renovate stage below:
stages {
		stage('Renovate') {
          	
          	environment {
    			ARTIFACTORY_ACCESS = credentials("renovate.bot.artifactory")
    			BITBUCKET_PROJECT_ID_UPPER = "${BITBUCKET_PROJECT_ID}".toUpperCase()
              		// https://docs.renovatebot.com/self-hosted-configuration/#gitprivatekey
    			RENOVATE_GIT_PRIVATE_KEY = credentials("team.bot-private-renovate.key")
          	}
          
      		steps {
        		script{
        
        			dir("${env.WORKSPACE}/renovate") {
                                				
              
          				sh '''
          					docker login -u $ARTIFACTORY_ACCESS_USR -p $ARTIFACTORY_TOKEN ZZZ.YYY.com
          					docker run \
    						    --rm \
            					-e ARTIFACTORY_ACCESS_USR=$ARTIFACTORY_ACCESS_USR \
                    				-e ARTIFACTORY_TOKEN=$ARTIFACTORY_TOKEN \
						-e BITBUCKET_ACCESS_PSW=$BITBUCKET_ACCESS_PSW \
            					-e BITBUCKET_ACCESS_USR=$BITBUCKET_ACCESS_USR \
            					-e BITBUCKET_PROJECT_ID=$BITBUCKET_PROJECT_ID_UPPER \
            					-e BITBUCKET_REPO=$BITBUCKET_REPO \
            					-e GITHUB_COM_TOKEN=$GITHUB_COM_TOKEN \
            					-e LOG_LEVEL=$LOG_LEVEL \
                            			-e RENOVATE_GIT_PRIVATE_KEY=$RENOVATE_GIT_PRIVATE_KEY \
                            			-v "./config.js:/usr/src/app/config.js" \
    						    ZZZ.YYY.com/thisisafolder/renovate/renovate:"${RENOVATE_CURRENT}" \
    						    --dry-run="false"
          				'''

### Logs (if relevant)

<details><summary>Logs</summary>

```
...
LOG_LEVEL=debug -e RENOVATE_GIT_PRIVATE_KEY=-----BEGIN PGP PRIVATE KEY 'BLOCK-----\n\nlTcAG<the full key is printed out here>=6XPt\n-----END' PGP PRIVATE KEY BLOCK----- -v ./config.js:/usr/src/app/config.js ZZZ.YYY.com/thisisafolder/renovate/renovate:37.388.1-full --dry-run=false
docker: invalid reference format: repository name (library/PGP) must be lowercase.
See 'docker run --help'.
...
```

</details>

**Comment by rarkins** [comment, marked as answer]
Docker is throwing that error, not renovate. Renovate doesn't even start yet.

**Reply by Apamobil** [reply]
Ok... I see. What is then the correct way to pass the key to Renovate that is running in a Docker container?

**Reply by rarkins** [reply]
It's potentially as simple as wrapping it in quotes, but we don't provide technical support for the Docker CLI here, only Renvoate

**Reply by Apamobil** [reply]
Adding quotes like this 

`-e RENOVATE_GIT_PRIVATE_KEY="${RENOVATE_GIT_PRIVATE_KEY}"`

did the trick! Thank you for the support! :)
